.github/workflows/cache-tests.yml

```diff
@@ -0,0 +1,61 @@
+name: cache-unit-tests
+on: push
```

Comment:
I would recommend path filters here. Certain things like `.md` files can be ignored: https://github.com/actions/toolkit/blob/83dd3ef0f1e5bc93c5ab7072e1edf1715a01ba9d/.github/workflows/unit-tests.yml#L7